### PR TITLE
replace version.h macros with a global object

### PR DESCRIPTION
**Pull request by arogge**
Previously every Bareos version built had its version number set as a
compile definition using -DVERSION. If neccessary the version number
had been extracted from version.h using cmake.

This patch now introduces struct BareosVersion with its single instance
kBareosVersion defined in version.c.
That struct now contains all version information. Consumers just link
against cmake target version-obj and will be able to access the
information.

The Bareos version is now configured in BareosVersion.cmake in both core
and webui.

### CODE REVIEW

File: core/src/lib/version.c [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  "\n"                                                           \
+  "Copyright (C) 2013-%s Bareos GmbH & Co. KG\n"                 \
+  "Copyright (C) %d-2012 Free Software Foundation Europe e.V.\n" \
+  "Copyright (C) 2010-2017 Planets Communications B.V.\n"        \
+  "\n"                                                           \
+  "Version: %s (%s) %s %s %s\n"                                  \
+  "\n"
+
+
+static void formatCopyright(char* out, size_t len, int FsfYear)
```
Comment (franku) [inline core/src/lib/version.c]:
```suggestion
static void FormatCopyright(char* out, size_t len, int FsfYear)
```

File: core/src/lib/version.c [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+{
+  snprintf(out, len, BAREOS_COPYRIGHT_MESSAGE, kBareosVersion.ServicesMessage,
+           kBareosVersion.Year, FsfYear, kBareosVersion.Full,
+           kBareosVersion.Date, HOST_OS, DISTNAME, DISTVER);
+}
+static void printCopyright(FILE* fh, int FsfYear)
```
Comment (franku) [inline core/src/lib/version.c]:
```suggestion
static void PrintCopyright(FILE* fh, int FsfYear)
```

File: core/src/lib/version.h [hunk 3]
```diff
@@ -53,6 +21,32 @@
    02110-1301, USA.
 */
 
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct BareosVersion {
```
Comment (franku) [inline core/src/lib/version.h]:
```suggestion
struct BareosVersionStrings {
```

File: core/src/lib/version.c [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+{
+  fprintf(fh, BAREOS_COPYRIGHT_MESSAGE, kBareosVersion.ServicesMessage,
+          kBareosVersion.Year, FsfYear, kBareosVersion.Full,
+          kBareosVersion.Date, HOST_OS, DISTNAME, DISTVER);
+}
+
+const struct BareosVersion kBareosVersion = {
```
Comment (franku) [inline core/src/lib/version.c]:
```suggestion
const struct BareosVersionStrings kBareosVersionStrings = {
```

File: core/src/qt-tray-monitor/tray-monitor.cc [hunk 5]
```diff
@@ -44,11 +44,10 @@ static QCoreApplication* app = nullptr;
 
 static void usage()
 {
-  QString out;
-
-  out = out.sprintf(
-      _(PROG_COPYRIGHT
-        "\nVersion: %s (%s) %s %s %s\n\n"
+  char copyright[500];
```
Comment (franku) [inline core/src/qt-tray-monitor/tray-monitor.cc]:
```suggestion
  std::vector<char> copyright{1024};
```

File: core/src/qt-tray-monitor/tray-monitor.cc [hunk 6]
```diff
@@ -44,11 +44,10 @@ static QCoreApplication* app = nullptr;
...
+  kBareosVersion.formatCopyright(copyright, 500, 2004);
```
Comment (franku) [inline core/src/qt-tray-monitor/tray-monitor.cc]:
```suggestion
  kBareosVersion.formatCopyright(copyright, copyright.size(), 2004);
```

File: core/src/qt-tray-monitor/tray-monitor.cc [hunk 7]
```diff
@@ -59,8 +58,7 @@ static void usage()
         "        -xs         print configuration file schema in JSON format "
         "and exit\n"
         "        -?          print this message.\n"
-        "\n"),
-      2004, VERSION, BDATE, HOST_OS, DISTNAME, DISTVER);
+        "\n")).arg(copyright);
```
Comment (franku) [inline core/src/qt-tray-monitor/tray-monitor.cc]:
```suggestion
        "\n")).arg(copyright.data());
```

File: core/src/stored/label.cc [hunk 8]
```diff
@@ -586,8 +586,9 @@ void CreateVolumeLabel(Device* dev, const char* VolName, const char* PoolName)
     dev->VolHdr.HostName[0] = 0;
   }
   bstrncpy(dev->VolHdr.LabelProg, my_name, sizeof(dev->VolHdr.LabelProg));
-  sprintf(dev->VolHdr.ProgVersion, "Ver. %s %s", VERSION, BDATE);
-  sprintf(dev->VolHdr.ProgDate, "Build %s %s", __DATE__, __TIME__);
+  sprintf(dev->VolHdr.ProgVersion, "Ver. %s %s", kBareosVersion.Full,
```
Comment (arogge) [inline core/src/stored/label.cc]: Make sure that VolHdr.ProgVersion does not exceed 50 chars.

**Pull request merged**